src/assets.rs

```diff
@@ -319,6 +344,22 @@ mod tests {
                 .get_syntax(None, &mut opened_input, &self.syntax_mapping);
             syntax.name.clone()
         }
+
+        fn syntax_is_same_for_inputkinds(&self, file_name: &str, content: &str) -> bool {
+            let as_file = self.syntax_for_real_file_with_content_os(file_name.as_ref(), content);
+            let as_reader = self.syntax_for_file_with_content_os(file_name.as_ref(), content);
+            let consistent = as_file == as_reader;
+            // TODO: Compare StdIn somehow?
```

Comment:
maybe we can (later) get rid of `StdIn` and move it to `Input::from_reader`?

Reply:
We will need to make a couple changes to `InputDescription` to allow for custom descriptions, but that would be a good idea. We likely would also get rid of the theme preview file input in the same PR.